server/enterprise/message_export/global_relay_export/global_relay_export.go

```diff
@@ -176,52 +178,70 @@ func addToExports(rctx request.CTX, allExports map[string][]*ChannelExport, gene
+	for _, start := range post.AttachmentStarts {
+		attachmentStarts = append(attachmentStarts, start.FileInfo)
 	}
-
-	// Create a new ChannelExport if it would be too many bytes to add the post
-	fileBytes := fileInfoListBytes(post.Attachments)
+	fileBytes := fileInfoListBytes(attachmentStarts)
 	postBytes := fileBytes + msgBytes
+	// NOTE: This is only a rough estimate -- we're not including the txt or html portion of the email...
+	attachmentsAloneTooLargeToSend := fileBytes > MaxEmailBytes // Attachments must be removed from export, they're too big to send.
+	if attachmentsAloneTooLargeToSend {
+		postBytes -= fileBytes
+	}
 	postTooLargeForChannelBatch := channelExport.bytes+postBytes > MaxEmailBytes
-	postAloneTooLargeToSend := postBytes > MaxEmailBytes // Attachments must be removed from export, they're too big to send.
 
-	if postAloneTooLargeToSend {
+	if attachmentsAloneTooLargeToSend {
 		attachmentsRemovedPostIDs = append(attachmentsRemovedPostIDs, *post.PostId)
 	}
 
 	// new "batch"
-	if postTooLargeForChannelBatch && !postAloneTooLargeToSend {
-		channelExport = &ChannelExport{
-			ChannelId:   genericChannel.ChannelId,
-			ChannelName: genericChannel.DisplayName,
-			ChannelType: genericChannel.ChannelType,
-			StartTime:   genericChannel.StartTime,
-			EndTime:     genericChannel.EndTime,
-			// we can't preallocate sizes here because we don't know how many will be in this "batch"
-			Messages:        make([]Message, 0),
-			Participants:    make([]ParticipantRow, 0),
-			numUserMessages: make(map[string]int),
-			uploadedFiles:   make([]*model.FileInfo, 0),
-			bytes:           0,
-		}
+	if postTooLargeForChannelBatch {
+		channelExport = genericChannelToChannelExport(genericChannel)
 		allExports[*post.ChannelId] = append(allExports[*post.ChannelId], channelExport)
 	}
 
-	// For now we're not exporting deleted messages (MM-62059), but we ARE exported the deleted files...
-	if post.UpdatedType != shared.Deleted {
-		addPostToChannelExport(rctx, channelExport, post)
-	}
+	addPostToChannelExport(rctx, channelExport, post)
 
 	// if this post includes files, add them to the collection
-	for _, fileInfo := range post.Attachments {
-		addAttachmentToChannelExport(channelExport, post, fileInfo, postAloneTooLargeToSend)
-	}
+	addAttachmentsToChannelExport(channelExport, post, post.AttachmentStarts, post.AttachmentDeletes, attachmentsAloneTooLargeToSend)
 	channelExport.bytes += postBytes
 	return attachmentsRemovedPostIDs
 }
 
+// addChannelWithNoPostsToExports is for channels that don't have posts, but we exported it anyway, so we need to track
+// it. It has joins and leaves.
+func addChannelWithNoPostsToExports(rctx request.CTX, allExports map[string][]*ChannelExport,
+	genericChannel shared.ChannelExport) {
+	if _, present := allExports[genericChannel.ChannelId]; !present {
+		// we found a new channel
+		allExports[genericChannel.ChannelId] = []*ChannelExport{genericChannelToChannelExport(genericChannel)}
+	}
+}
```

Comment:
I think this hits the point (in my opinion) where a method is a little too much. We are literally doing a conditional map access/assignment.

Reply:
Ah yes, I think I might have been too aggressive there thinking it might be used elsewhere. Fixed.